pkg/operator/ceph/controller/finalizer.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ObjectFinalizer is the name of the finalizer
+	ObjectFinalizer = "finalizer.ceph.io"
+)
+
+// Contains checks if an item exists in a given list.
+func Contains(list []string, s string) bool {
+	for _, v := range list {
+		if v == s {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Remove removes any element from a list
+func Remove(list []string, s string) []string {
```

Comment:
Per previous comment, seems like `Contains()` and `Remove()` could just be implementation details inside the `Add/RemoveFinalizer()` methods. If that works out, these could be lower case (private) methods.

Reply:
This can work for `Remove` but not for `Contains()`

Reply:
Nevermind my last comment, I reworked it.